daemon/rpc/swap_out.go

```diff
@@ -3,35 +3,37 @@ package rpc
 import (
 	"context"
 	"crypto/rand"
-	"crypto/sha256"
-	"encoding/hex"
+	"fmt"
 
 	"github.com/40acres/40swap/daemon/database/models"
 	"github.com/40acres/40swap/daemon/money"
 	"github.com/40acres/40swap/daemon/swaps"
+	"github.com/lightningnetwork/lnd/lntypes"
 	log "github.com/sirupsen/logrus"
 )
 
-func (server *Server) CreateSwapOut(ctx context.Context, claimPubKey string, amountSats money.Money) (*swaps.SwapOutResponse, error) {
+func (server *Server) CreateSwapOut(ctx context.Context, claimPubKey string, amountSats money.Money) (*swaps.SwapOutResponse, *lntypes.Preimage, error) {
 	log.Info("Creating swap")
 
-	preimage := make([]byte, 32)
-	_, _ = rand.Read(preimage)
-	hash := sha256.New()
-	hash.Write(preimage)
-	preimageHash := hash.Sum(nil)
+	preimageBytes := make([]byte, 32)
+	_, _ = rand.Read(preimageBytes)
+
+	preimage, err := lntypes.MakePreimage(preimageBytes)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not create preimage: %w", err)
+	}
 
 	swapRequest := swaps.CreateSwapOutRequest{
 		Chain:        models.Bitcoin,
-		PreImageHash: hex.EncodeToString(preimageHash),
+		PreImageHash: preimage.Hash().String(),
```

Comment:
Mmmm, didnt we talk about HASH160 hashing? is this working?

Reply:
Yep, it is working. Since from the backend we expect sha256 and then here we only do hash160: https://github.com/40acres/40swap/blob/6a8d1d5b3791e303492bbee95ce24d1b8b4a727a/server-backend/src/bitcoin-utils.ts#L36-L37